### ISSUE TITLE
Make config file discovery simple, like pycodestyle

### ISSUE THREAD

**Issue by nlfiedler** [PyCQA/pydocstyle#278]
I use `pycodestyle` and `pydocstyle` with Atom (via `pylama`), so I am not invoking either tool myself. As such, configuring them is tricky. Fortunately, `pycodestyle` will read the configuration file from `~/.config/pycodestyle`, regardless of how the tool is run. However, there is nothing similar for `pydocstyle`. What's more, it's difficult to know how/where the tool is being run via `pylama`, so I don't know where to put the pydocstyle config. If I could put it in `~/.config` then it would be much simpler. Plus it would be consistent with the other tool.

**Comment by Nurdok** [comment]
pydocstyle `2.0.0` added a `--config=<config-file>` option which uses the given config file and disables the complex config discovery (explained below). You could pass `--config=~/.config/pydocstyle` and get the same behavior as pycodestyle.

The normal pydocstyle configuration discovery works by finding configuration files in the project where the files are located. So when you run `pydocstyle` on a file, it looks for configuration files next to it and searches up in the file system. This is because different projects can use different conventions. We chose this scheme to optimize for the process of converting a non-complying project to conform to a given convention. So, if you want to convert a huge legacy project to have consistent conventions, you could do it directory-by-directory using different configuration files in a comfortable way. I realize this is not the case for every project and that's why the `--config` option was added.

**Comment by nlfiedler** [comment]
I am using the Atom editor, with the linter-python package, and it invokes pylama. I don't have the kinda of control that I would like in order to invoke pydocstyle as you describe. I am _not_ asking you to support a particular editor just because I asked. What I am saying is that there are these two tools, pycodestyle and pydocstyle, which are ostensibly related and very similar, and yet behave very differently when it comes to configuration. As such, using the one (pycodestyle) is very, very convenient, while the other is not. You may do with that what you will.

**Comment by sigmavirus24** [comment]
@nlfiedler let me give you this bit of feedback: As one of the maintainers of pycodestyle, I want to kill user-global configuration files. They are the root of so many problems and not worth the effort I have put in to keep them functional in Flake8 when rewriting it for v3.0

**Comment by Nurdok** [comment]
@sigmavirus24 that's really interesting! Can you expand on the reasons?

**Comment by sigmavirus24** [comment]
Flake8 (and pycodestyle) have had more "When I run Flake8 on this project with a project config like (pasted config) I get X, Y, and Z errors but my coworker doesn't and I don't know where they're coming from" and the answer is always "User-global config".

Some people expect that the user-global config will just be ignored when there's a project-local one. Other people have wildly different expectations of how things work. And every other tool that has this (even outside the Python community) has vastly different behaviours. 

In other words, there's no one right, obvious, or intuitive way of handling them. They provoke confusion and gnashing of teeth. They're clumsy. Their location varies based on operating system convention. People forget they exist and miss the fact in verbose output that it's being read. It's misery for all involved.

This thread is provoking me to add deprecation warnings to pycodestyle and Flake8 right this second when we find and read a user-global config file so we can kill them soon.
